MalwareAnalyzer/views/apkid.py

```diff
@@ -1,26 +1,41 @@
 # -*- coding: utf_8 -*-
 import os
+import subprocess
+import json
 import logging
+import shutil
 from django.conf import settings
 from MobSF.utils import filename_from_path
 
 logger = logging.getLogger(__name__)
 
 
-def apkid_analysis(app_dir, apk_file):
+def apkid_analysis(app_dir, apk_file, apk_name):
     """APKiD Analysis of DEX files"""
     apkid_res = {}
     if not settings.APKID_ENABLED:
         return apkid_res
     if not os.path.exists(apk_file):
         logger.error("APKiD - APK not found")
         return {'error': True}
+    dest_dir = os.path.join(app_dir, "apkid/")
+    os.mkdir(dest_dir)
+    shutil.copy2(apk_file, dest_dir)
     from apkid import apkid
     from apkid import __version__ as apkid_ver
     logger.info("Running APKiD %s on: %s", apkid_ver,
                 filename_from_path(apk_file))
-    result = apkid.scan_apk(apk_file, 30, True)
+    result = apkid.scan(dest_dir, 30, True, dest_dir, True)
+    temporary_apk = dest_dir +  filename_from_path(apk_file)
+    report = dest_dir +  filename_from_path(apk_file) +  "_apkid.json"
+    with open(report, 'r') as f:
+        result = json.load(f)
     for filep, res in result.items():
-        dex_file = filep.rsplit("!", 1)[1]
+        try:
+            dex_file = filep.rsplit("!", 1)[1]
+        except Exception:
```

Comment:
Why try and exception?
You can do an if else to check for. Apk or. Dex right?

Reply:
yep I could :)